### DISCUSSION TITLE
:(

### DISCUSSION THREAD

**Discussion by 4lt3r3go** [Gerschel/sd-web-ui-quickcss#3] [General]
(EDIT: SOLVED)
unfortunatly i had to uninstall this amazing extention, too many errors were popping up in the consolle
bye bye cute layout :( 
![image](https://user-images.githubusercontent.com/44883288/210607345-d184c338-58cd-492f-9a1c-e91f4fc1080f.png)

**Comment by Gerschel** [comment]
I'm not getting a single error. But I haven't updated to webui yet.  
I'd like to know more.  
Were you on a recent update of this extension?

**Comment by 4lt3r3go** [comment]
> I'm not getting a single error. But I haven't updated to webui yet. I'd like to know more. Were you on a recent update of this extension?

yup. installed today

**Comment by Gerschel** [comment]
You might need to update the css.  
I had the dynamic branch, and the master branch.  
I will also be updating the css after I get through some backlog.  

I merged them both in to master.  
If it's caused because I changed how the dynamic reads the file, I think I know where to add an error check to give proper hinting.  

The script looks for these, the word `quickcss_target`, `/*ENDCOLORPICKERS*/`, and `/*BREAKFILEREADER*/`.  
The rule header such as `root, *, quickcss_target` in this example, needs to be on the same line.
Must use hex values for colors, no extra blank lines, no extra comments.
The ones after colorpickers zone is for sliders, they must have no units specified, units can be added in other rules.
Extra variables that you don't want to have adjustable go after `/*BREAKFILEREADER`.


![image](https://user-images.githubusercontent.com/9631031/210627455-1e7442b4-fdcd-47d6-9072-d4201d198f8d.png)

This then gives colorpickers and sliders in the extension, for modifying and saving.
![image](https://user-images.githubusercontent.com/9631031/210628408-7484cfce-6ea8-4ef8-a2c4-28432503e83a.png)

If you don't need this. Remove `quickcss_target`, and it will treat it like a normal stylesheet.  
There are iterations of the stylesheet that were upgrading as this was being made.  
Even the one labeled neon.css in the repository is not the same neon that was applied a few days ago.  

If it's one of these neons that was for a prior read method, you can update it by reapplying.  
The current version works with these features, it has a jankiness, I threw in the demonstration of radius.  
It will be updated later today, since I updated the readme with some instructions instead of relying on the stylesheet to demonstrate the instructions.  

I'm going to thoroughly investigate, add some error handling, and make an update today.

**Comment by Gerschel** [comment]
TLDR: Just reapply the stylesheet from the repo, stylesheet was updated. Readjust your colors, save.

I think I know what was going on. In the css, it specified a different filename/location for logos.  
I updated the neon.css to @nacurutu https://github.com/Nacurutu latest.  
I made an alteration to his file that's worth noting.  
I changed the url parameters to the location where the script names it as logo.png and favicon.svg, I may need to update how this works. Change the save location to a static folder in the extension. Hot swap as needed.  Static folder in root makes more sense, as it would work whether or not extension is installed.
  
I added `.gradio-container` to `* { border-radius ...}, so it doesn't affect the radius outside the modal.  
I was getting errors too, mine were logo.png is not found at ..., this is because I don't have a logo.png in that location, they're still the default names as downloaded from github.

**Comment by Nacurutu** [comment]
@4lt3r3go everything is working for me with no errors.

Also you can try with the files on this repo:

https://github.com/Nacurutu/Dark-Themes-SD-WebUI-Automatic1111

Im gonna try to keep the files updated as much as I can.

**Comment by Nacurutu** [comment]
@4lt3r3go did a test with your color palette, didnt get any visual or console errors.

![image](https://user-images.githubusercontent.com/18109442/210680850-08672f1a-ffb4-4179-b44e-53f717237ffd.png)

Try replacing the code of your css file with this one:

https://pastebin.com/yivVAuVX

Let me know.. :)

**Comment by 4lt3r3go** [comment]
thanks guys, i wanst expecting so many attentions and technical support ❤️
However i'm having other issues now that doesnt allow me to use SD normally at all 
( damn me, i should remove that --git pull from the bat)
Once i fix this bs i'll for sure give to this extention another try.
**QUESTION:**
Does this extention allow somehow to customize positions of buttons and sliders? or dimensions?
this would be awesome

**Comment by Nacurutu** [comment]
> thanks guys, i wanst expecting so many attentions and technical support ❤️ 

More than welcome!

> However i'm having other issues now that doesnt allow me to use SD normally at all ( damn me, i should remove that --git pull from the bat) Once i fix this bs i'll for sure give to this extention another try.

I have 2 "SD webUI" installed, a simple one without scripts/extensions/custom models where I use github to pull updates and test the app, and, if everything goes fine and the update didnt break the app, I copy and paste all the files to the "second one". With that, I avoid breaking the one and only with a broken update. :wink:

I also made a second webui-user.bat called webui-user-Personal.bat where I have the COMMANDLINE_ARGS applied, with that I also avoid any mistake/override/modification/issue with the original webui-user.bat when I copy all the files from "SD Vanilla" to my "SD Tuned". 

Right now im having a bug with the send to img2img button, I think there is an error with the app from the last update and maybe will get a fix soon. The send to inpaint works fine. (just in case you update the the app and get the same bug). 

>  **QUESTION:** Does this extention allow somehow to customize positions of buttons and sliders? or dimensions? this would be awesome

Positions, I dont know if that is possible just modifying the css file, a css file is just for the style. 
I think in order to customize positions, should be done on the python file, however, that is going to break updates, and everytime Automatic1111 update the app, you will need to make all the changes again. Someone correct me if im wrong. 

Dimentions are possible, however could be a little pain in the butt because it depends on what you need to change and specifiy it on the file. What do you have in mind? what dimensions you need to change? 




EDIT: I have updated again your css file and added the new modifications I made to mine, some code fix and nothing more.

https://pastebin.com/yivVAuVX

**Comment by Gerschel** [comment]
@4lt3r3go 
```bash
git fetch --all
git reset --hard origin/master
```
Git pull is two commands in one, it's a git fetch and a git merge.
In this one, you do the fetch, then say fuck it, replace the files anew.  
This wont make you redownload your models, just the same files you were trying to get with git pull.

If it's just complaining about the webui.py because you made a change. 

`git checkout master -- webui.py`

This means, undo my local changes on this file. Then a git pull would work.

Just about everyone has had a problem with recent updates, because the way others merged before committing.  

@Nacurutu 
There's ways to get tricky with css to force position. At the worse, there's javscript code like this. 
I played with it. I have a code in the preset manager that's in your copy right now.  
Copy this and place in your browser devtools console.
```javascript
let mover = new PresetManagerMover("txt2img_preset_manager_accordion", "quicksettings", "")
mover.move('beforebegin')
```

Then to move it back
```javascript
mover.updateTarget('txt2img_generation_info_button')
mover.move('afterend')
```

I got a pull request merged in to the webui that allows tabbed components in quicksettings area, just for this.

**Comment by Nacurutu** [comment]
Awesome... 

As always, I didnt knew that could be possible from your preset extension and/or css.

And yep, I know that now we can organize quicksettings depending on the order you want, didnt know was ur idea and code, good job!

**Comment by Gerschel** [comment]
The mover was a test to see how it handled in quicksettings. They have a javacript function get_uiTabs, which targetted my extension when I had it up top, instead of their tabs. I just got an event working that would move it automatically, it was tricky. https://github.com/gradio-app/gradio/discussions/2932#discussioncomment-4598826 
So now I need to handle hiding on tab switch, or change contents on tab switch. I had a local version that worked on every tab for every component, but required adding code to `ui.py`
I'm trying to work around turning my preset manager into built-in until he gives a license I can agree with.
We'll take this discussion back to discussions.

**Comment by Nacurutu** [comment]
Sounds awesome...

What if you try to make a pull request adding that code to the main ui.py?

Maybe they add the code and you can add the function to the presets utils or quick css extension. Should be great to be able to move everything as you wish.

**Comment by 4lt3r3go** [comment]
> What do you have in mind?

i'm just bored to go find the denoise slider in the bottom of the page, and the latest update added even more space between the image and that slider.. i'm consuming my mouse wheel XD



> I have updated again your css file

TY.



> There's ways to get tricky with css to force position.

lookign forward for this! i really wish to customize all stuff positions



Well... in the end Auto1111 repo is moved away from github today so i dunno wtf how all this gonna evolve..

**Reply by Gerschel** [reply]
When the made the ui changes, they added a field in settings, that let you rearrange groups of components.  
Settings tab,
User Interface tab on left. 
Near bottom.  

Just make sure have all included, and dont include the same things twice.  

![image](https://user-images.githubusercontent.com/9631031/210842831-693be3da-c3b4-4da0-a846-83f7b0781d83.png)

**Reply by Nacurutu** [reply]
ufff... thats amazing..!

Thanks for pointing that..! do you know if that can be addressed from the css file? like to stablish the order automatically from the css?

Can we write inputs directly from the css? or maybe, can u edit that field in settings from your extension?

**Reply by 4lt3r3go** [reply]
> they added a field in settings

yeah i already tryed that, by right click and inspect browser stuff i tryed find the "denoise strenght" but didnt worked for me.
thats the main one i want to move. But its ok in the end, this new CSS styles make text smaller so now i can see the slider without scroll down to the bottom of the page (that was so annoy)

**Reply by Nacurutu** [reply]
Just change "cfg" position,

![image](https://user-images.githubusercontent.com/18109442/210870768-641c2d08-f9ae-41e4-924c-043443d70e0e.png)

![image](https://user-images.githubusercontent.com/18109442/210871060-71d8bf65-6b51-4fa1-b11c-03632a651325.png)

"cfg" will move CFG Scale and Denoising strength.

**Comment by Nacurutu** [comment]
Aparently, Automatic1111 Github account got suspended.

However, he moved the repo to Gitgud.

Here's the gitgud link to the WebUI

[Gitgud Automatic1111 Stable Difussion WebUI](https://gitgud.io/AUTOMATIC1111/stable-diffusion-webui)

Here's the reddit where he announced the migration:

[Reedit post](https://www.reddit.com/r/StableDiffusion/comments/103vsm2/webuis_new_home/)

So, now you need to clone the repo from Gitgud. I dnt know if he is going to use github again.

Note 1: you can download the Source code from Gitgud and paste the files on your SD WebUI root folder. It will update the app.

Note 2: If you have installed an extension that was hosted on Automatic1111's repos, go to your extensions folder, Copy them to a different folder outside extensions (like a backup folder) and delete them from the extensions folder, otherwise the Extensions / Check for updates will break. 

I know that

> stable-diffusion-webui-aesthetic-gradients
> stable-diffusion-webui-wildcards

were on his repo.

> i'm just bored to go find the denoise slider in the bottom of the page, and the latest update added even more space between the image and that slider.. i'm consuming my mouse wheel XD

Lets wait a bit for @Gerschel, I know he will manage a way to do it and to include the repositioning on one of his Extensions.


![image](https://media.giphy.com/media/3o72FfM5HJydzafgUE/giphy.gif)

**Comment by Nacurutu** [comment]
@4lt3r3go I think the Automatic1111 Github is back online

https://github.com/AUTOMATIC1111/stable-diffusion-webui

**Reply by Gerschel** [reply]
It's back up. It was because there were links in the wiki textual embeddings to NSFW and used other artists (both controversies).

**Reply by Nacurutu** [reply]
yep, it is..!!

**Comment by IkariDevGIT** [comment]
i get many errors with the new retrowave style

**Reply by Gerschel** [reply]
Which erors were you getting?

**Comment by Nacurutu** [comment]
> i get many errors with the new retrowave style

retrowave is still in beta, it's a css that @ParityError is still working on it. I think it was uploaded for testing some things.

You can try this ones:

[Dark Themes SD WebUI Automatic1111](https://github.com/Nacurutu/Dark-Themes-SD-WebUI-Automatic1111)

and you can change the color palette using the extension. 
Im updating my css files often. @Gerschel adds them every few days.

**Comment by 4lt3r3go** [comment]
> You can try this one

i wonder if there's a place were to browse for different styles

**Comment by Nacurutu** [comment]
Well, im on it. Im one of the persons making styles for the extension, however, first im making all the possible changes to one, clean the code as much as I can and then, im going to start making different styles with great color palettes.

Patience...
![image](https://media.giphy.com/media/xf20D8HzvTQzu/giphy.gif)

I know @catppuccin made 4 css files, you can apply them from the Extension, however, they cant be modified using the Extension. Catppuccino asked to not add them to the Extension's repo, so the only way is to download them from there:

[Catppuccin Flavors for stable diffusion](https://github.com/catppuccin/Stable-Diffusion-Web-UI)

We cant add the parameters to make them compatible with the color picker, but you can if you want.

**Comment by Gerschel** [comment]
@Nacurutu Kryptortio updated element id's and created forcing unique element id's on scripts that don't use them.  
https://github.com/AUTOMATIC1111/stable-diffusion-webui/compare/42fcc79...01a1fee  
Then automatic modified it.  
https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/f8d0cf6a6ec4911559cfecb9a9d1d46b547b38e8  

I go through every commit so I can keep an updated snapshot in my head, see if it breaks anything I work on.  

I dont think this affects how the css rules are being written.  It's just worth noting.  
But this means I need to update the preset manger into elemenet id's, and write a script to update person presets into the new targets. So I'm going to be stuck on that for a bit.

**Reply by Nacurutu** [reply]
not a problem, take your time..!

**Comment by 4lt3r3go** [comment]
Ty everyone, now this extention works flawless
(i added SOLVED to the main post)
. I can finally enjoy new CSS styles!
now we need a hub place were to upload them.. like a sort of Civitai for styles :P

**Comment by AdrianoAz** [comment]
:( #4 - error loading script: quickcss.py lines: 205, 13, 383, 33, 327 the system cannot find se-web-ui-quickcss\\effects. 
thanks anyway

**Reply by Gerschel** [reply]
For future reference. Could I get better information from the traceback.  
It doesn't need to include your folder structure. But the file name just to the left of the line number, or  function name to the right, would dramatically help.  

That way I'm not guessing the file the line number is for.

**Reply by AdrianoAz** [reply]
ok, sorry, thanks

**Comment by Exozze** [comment]
Error loading script: quickcss.py
Traceback (most recent call last):
  File "C:\Users\egorv\stable-diffusion-webui\modules\scripts.py", line 205, in load_scripts
    module = script_loading.load_module(scriptfile.path)
  File "C:\Users\egorv\stable-diffusion-webui\modules\script_loading.py", line 13, in load_module
    exec(compiled, module.__dict__)
  File "C:\Users\egorv\stable-diffusion-webui\extensions\sd-web-ui-quickcss\scripts\quickcss.py", line 383, in <module>
    tab = MyTab(basedir)
  File "C:\Users\egorv\stable-diffusion-webui\extensions\sd-web-ui-quickcss\scripts\quickcss.py", line 33, in __init__
    self.effects_list = self.get_files(self.effects_folder, file_filter=["quickcss.js", "utility.js", "background.js"], split=True)
  File "C:\Users\egorv\stable-diffusion-webui\extensions\sd-web-ui-quickcss\scripts\quickcss.py", line 327, in get_files
    focus_list = [file_name if not split else os.path.splitext(file_name)[0] for file_name in os.listdir(folder) if os.path.isfile(os.path.join(folder, file_name)) and file_name not in file_filter]
FileNotFoundError: [WinError 3] the system cannot find: 'C:\\Users\\egorv\\stable-diffusion-webui\\extensions\\sd-web-ui-quickcss\\effects'

**Reply by Gerschel** [reply]
I'm on it.

**Reply by Gerschel** [reply]
@Exozze @AdrianoAz 
It's patched now.
Yeah, I have the same files in that folder, so it didn't end up throwing me the error. That was going to be the home for storing the effects as they are built.
